### DISCUSSION TITLE
pulsar.sh looking for binary in tmp folder

### DISCUSSION THREAD

**Discussion by arifsaeed** [pulsar-edit/.github#283] [Hacking Pulsar]
I tried to follow the instructions as best as i could in https://pulsar-edit.dev/docs/launch-manual/sections/core-hacking/#building-pulsar.  I did clone pulsar to a non standard location.  when i try to run the command: 

**pulsar --dev <path-to-open>**

it cant find the binary, but looks like its looking in location /opt/pulsar or tmp/pulsar-build but i cant find the binary in those locations or anywhere.  

are you able to provide any help on how to debug this problem would be greatly appreciated 

ive attached a screenshot of the setup.  
![pulsarfoldersetup](https://github.com/user-attachments/assets/eece3c38-2562-40bf-b640-2be63a28d0f1)

also running yarn start seems to start the application fine.

**Comment by savetheclocktower** [comment]
I believe [#1066](https://github.com/pulsar-edit/pulsar/pull/1066) will fix this. I hope to get it landed in time for the mid-September release of 1.121.

Until then, here are the possible workarounds that come to mind:

* Start Pulsar with `yarn start`, then use the “Open in Dev Mode…” menu item to open further projects. (It should be under _View > Developer_).
* Grab the `pulsar.sh` script from [#1066](https://github.com/pulsar-edit/pulsar/pull/1066) and put it somewhere temporary. That script will fail to guess the location of your Pulsar installation, but it will use the `PULSAR_PATH` environment variable if it's defined. So you could do `export PULSAR_PATH="~/Documents/design/Prod/pulsar"` in a shell startup file or something (or just add it to the top of that `pulsar.sh`). Once 1.121 comes out you can revert to using the standard script.
